src/groups/mqb/mqbblp/mqbblp_relayqueueengine.cpp

```diff
@@ -1682,17 +1676,472 @@ bool RelayQueueEngine::subscriptionId2upstreamSubQueueId(
         const Routers::SubscriptionIds::SharedItem itId =
             d_queueState_p->routingContext().d_groupIds.find(subscriptionId);
         if (!itId) {
+            BMQ_LOGTHROTTLE_ERROR()
+                << "#QUEUE_UNKNOWN_SUBSCRIPTION_ID "
+                << "Remote queue: " << d_queueState_p->uri()
+                << " (id: " << d_queueState_p->id()
+                << ") received a PUSH message for guid " << msgGUID
+                << ", with unknown Subscription Id " << subscriptionId;
+
             return false;  // RETURN
         }
         if (itId->value().d_priorityGroup == 0) {
             // The 'd_queueContext.d_groupIds' may contain new ids for which
             // configure response is not received yet.
+
+            BMQ_LOGTHROTTLE_ERROR()
+                << "#QUEUE_UNKNOWN_SUBSCRIPTION_ID "
+                << "Remote queue: " << d_queueState_p->uri()
+                << " (id: " << d_queueState_p->id()
+                << ") received a PUSH message for guid " << msgGUID
+                << ", with unconfigured Subscription Id " << subscriptionId;
+
             return false;
         }
         *subQueueId = itId->value().upstreamSubQueueId();
     }
     return true;
 }
 
+unsigned int
```

Comment:
This function is the actual logic, and it is clear as day what's going on.  This is the major beneficiary of the refactor, code clarity-wise.